src/activity_actor.cpp

```diff
@@ -2679,8 +2679,67 @@ void reload_activity_actor::finish( player_activity &act, Character &who )
     }
 }
 
-void reload_activity_actor::canceled( player_activity &act, Character &/*who*/ )
+void reload_activity_actor::canceled( player_activity &act, Character &who )
 {
+    item &reloadable = *reload_targets[0];
+    item &ammo = *reload_targets[1];
+    const bool ammo_uses_speedloader = ammo.has_flag( flag_SPEEDLOADER );
+    const bool reloadable_loads_one = reloadable.has_flag( flag_RELOAD_ONE );
+
+    const int num_loaded = ( act.moves_total - act.moves_left ) / ( act.moves_total / quantity );
+
+    if( can_reload() && !ammo_uses_speedloader && reloadable_loads_one && num_loaded > 0 ) {
+
+
+
+        const std::string reloadable_name = reloadable.tname();
+        // cache check results because reloading deletes the ammo item
+        const std::string ammo_name = ammo.tname();
+        const bool ammo_is_filthy = ammo.is_filthy();
+
+        if( !reloadable.reload( who, std::move( reload_targets[1] ), num_loaded ) ) {
+            add_msg( m_info, _( "Can't reload the %s." ), reloadable_name );
+            return;
+        }
+
+        if( ammo_is_filthy ) {
+            reloadable.set_flag( flag_FILTHY );
+        }
+
+        if( reloadable.get_var( "dirt", 0 ) > 7800 ) {
+            add_msg( m_neutral, _( "You manage to loosen some debris and make your %s somewhat operational." ),
+                     reloadable_name );
+            reloadable.set_var( "dirt", ( reloadable.get_var( "dirt", 0 ) - rng( 790, 2750 ) ) );
+        }
+
+        add_msg( m_neutral, _( "You insert %dx %s into the %s." ), num_loaded, ammo_name, reloadable_name );
```

Comment:
```suggestion
        add_msg( m_neutral, _( "You insert %1$dx %2$s into the %3$s." ), num_loaded, ammo_name, reloadable_name );
```
These are positional parameters to help the translators - another language may not order the words the same way.